metals/src/main/scala/scala/meta/internal/metals/Indexer.scala

```diff
@@ -130,21 +130,25 @@ final case class Indexer(
             scribe.info(s"Skipping reload with status '${status.name}'")
             Future.successful(BuildChange.None)
           case None =>
-            for {
-              userResponse <- workspaceReload().requestReload(
-                buildTool,
-                checksum,
-              )
-              installResult <- {
-                if (userResponse.isYes) {
-                  reloadAndIndex(session)
-                } else {
-                  tables.dismissedNotifications.ImportChanges
-                    .dismiss(2, TimeUnit.MINUTES)
-                  Future.successful(BuildChange.None)
+            if (clientConfig.autoImportBuilds == AutoImportBuildKind.All) {
```

Comment:
About the problem that @tgodzik has mentioned previously (potentially missed import in case of multiple build changes in quick succession), maybe a future improvement would be to always save the last "intent" to import (like a checksum?) and check if it has been changed in the meantime after the currently ongoing import is finished (poor man's event queue)? :slightly_smiling_face:

Reply:
Yep I was thinking of something like that, would be interested to look at what rust-analyzer does here. That said keen to keep this PR simple, this could be a future iteration